### PR TITLE
Fix find checkpoint

### PR DESCRIPTION
**Pull request by yrong**

### CODE REVIEW

**Review by claravanstaden — APPROVED** [review]
Nice, thanks!

File: relayer/relays/beacon/header/header.go [hunk 1]
```diff
@@ -555,7 +555,7 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
 	endIndex := startIndex + 1
```
Comment (yrong) [inline relayer/relays/beacon/header/header.go:555]: Btw: The CI tests failed for running too long,  what about `startIndex` is the last item of the ring buffer, will it cause the endless loop?

Comment (claravanstaden) [reply relayer/relays/beacon/header/header.go:555]: You're right! Fix here :) https://github.com/Snowfork/snowbridge/pull/1304

**Comment by yrong** [comment]
Closed in favor of https://github.com/Snowfork/snowbridge/pull/1304

**Pull request closed without merging**